controllers/managedosversionchannel_controller.go

```diff
@@ -18,36 +18,43 @@ package controllers
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"time"
 
 	elementalv1 "github.com/rancher/elemental-operator/api/v1beta1"
 	"github.com/rancher/elemental-operator/pkg/log"
 	"github.com/rancher/elemental-operator/pkg/syncer"
 	"github.com/rancher/elemental-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	errorutils "k8s.io/apimachinery/pkg/util/errors"
-	"k8s.io/client-go/rest"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/utils/pointer"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-const baseRateTime = 1 * time.Second
-const maxDelayTime = 256 * time.Second
-const defaultSyncTime = "1h"
+const (
+	baseRateTime        = 1 * time.Second
+	maxDelayTime        = 256 * time.Second
+	displayContainer    = "display"
+	minTimeBetweenSyncs = 10 * time.Second
```

Comment:
nit: I would increase the minTimeBetweenSyncs... 60 seconds maybe

Reply:
Yes probably, even it should not be that problematic or it should be configurable as current unit tests relay on it, now they wait 10 seconds only.

In any case I am not very happy having sort of constants and preventive constraints. With the current implementations this is required for two reasons:

* We are ignoring delete events of the Pod, hence when the Pod is deleted no reconcile loop is triggered. So if the pod is deleted being in Pending state we need a mechanism to actually move away from Pending. Because of that when the Pod is in pending or running state there is a polling mechanism to enqueue, that is to ensure it is not stuck in there and never enqueued again.

* The transition from Running to Succeeded only happens once, hence the reconcile loop is enqueued only once, if that reconcile loop is missed the synchronization never ends. For instance, it could happen we get a reconcile loop triggered by a succeeded state transition but the actual reconcile loop when pulling the pod it gets a cached one that is not yet succeeded. To solve that there is the above mentioned polling mechanism.

I am not very happy with it, but it is the only method I could think of being capable to react on missed events, by an active polling during the synchronization.

Reply:
Set to 60 by default